### ISSUE TITLE
Aggregating emissions species to CO2 equivalent by AR-x

### ISSUE THREAD

**Issue by danielhuppmann** [openclimatedata/openscm#70]
Side-thread to #52, related to #53 

**Is your feature request related to a problem? Please describe.**

For the IPCC SR15 analysis, I needed to aggregate various emissions species to aggregate CO2-equivalent, keeping in mind that the multipliers changed between IPCC assessment reports, and also working off varying level of detail reported by IAMs:
 - all P/HFC species individually, 
 - only aggregate PFC and HFC timeseries
 - only aggregate F-Gases timeseries (without specifying which source was used for the multipliers)
 - no data on any F-gases

**Describe the solution you'd like**

It would be great to have a function that takes emissions timeseries of varying level of detail and returns the CO2-equivalent aggregate, allowing to specify multiplier sources and warming potential duration. The feature should be also be flexible int he level of detail required (e.g, if NH3 is not provided, does the function return the aggregate of all available species or a timeseries of `nan`)

This function should be callable from the `pyam` package. This could be a first test case for the integration between an `IamDataFrame` and the openscm format.

**Describe alternatives you've considered**

An alternative would be to implement such a feature directly in `pyam` - but it will probably be useful to other climate modellers, too.

**Comment by swillner** [comment]
We could define this procedure for our standard set of parameters (#22), e.g. when reading aggregated values from `("Emissions")` and leave mulitplier sourcer and warming potential duration to `pint` contexts (we would such an aggregation in the low-level API, anyway, so it should be done there).
@danielhuppmann Are these parameters indepentent for different gases or just a handful of standard combinations for different gases?

**Comment by znicholls** [comment]
@danielhuppmann I've started on this in #140, is this the sort of test you need to pass? I am not really sure what exactly all your use cases require.

```python
@pytest.mark.parametrize(
    "metric_name,species,conversion",
    (
        ["AR4GWP100", "HFC32", 675],
        ["AR4GWP100", "SF6", 22800],
        ["AR4GWP100", "C2F6", 12200],
    )
)
def test_metric_conversion(metric_name, species, conversion):
    base = 1 * unit_registry("kg {} / yr".format(species))
    with unit_registry.context("AR4GWP100"):
        np.testing.assert_allclose(base.to("kg CO2 / yr"), conversion)
```

**Comment by danielhuppmann** [comment]
thanks @znicholls! Just worked through the beauty of the unit registry, I think that this is pretty much (or rather far more advanced than) what I what I had in mind...

We already have a function `df.convert_unit()` in `pyam`, see https://github.com/IAMconsortium/pyam/blob/381c4f6efccf1691feacff3f961b9b67645a422a/pyam/core.py#L626, which takes a mapping like `{'CH4': ('CO2', 25)}` and applies it to an `IamDataFrame`.

Would it be possible to add a function to the `UnitConverter` to return such a dictionary, e.g.,
```
mapping = openscm.UnitConverter(source='CH4', target='CO2', context='AR4GWP100').as_dict()
df.convert_unit(mapping)
```

**Comment by znicholls** [comment]
yep we can do something like that

**Comment by swillner** [comment]
Closed by #140
